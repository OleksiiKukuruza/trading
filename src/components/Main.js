import React from 'react';
import styled from 'styled-components';
import OrderList from './OrderList';
import MatchList from './MatchList';

const StyledMain = styled.div`
  display: flex;
  justify-content: space-between;

  > div {
    flex: 1;
  }
`;

const Main = ({ sellOrders, buyOrders, matches }) => (
  <StyledMain>
    <OrderList orders={buyOrders} title="Buy" />
    <OrderList orders={sellOrders} title="Sell" />
    <MatchList matches={matches} />
  </StyledMain>
);

export default Main;
