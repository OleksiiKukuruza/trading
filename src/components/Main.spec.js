import React from 'react';
import { shallow } from 'enzyme';
import Main, { StyledMain } from './Main';
import theme from '../theme';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('Main', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup(Main, {
      buyOrders: [{ id: 1, price: 123, quantity: 10 }],
      sellOrders: [{ id: 2, price: 233, quantity: 5 }],
      matches: [{ time: 123, sell: { id: 3, price: 50, quantity: 3 }, buy: { id: 4, price: 60, quantity: 6 } }]
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledMain', () => {
    const { wrapper } = setup(StyledMain, { theme });
    expect(wrapper).toMatchSnapshot();
  });
});
