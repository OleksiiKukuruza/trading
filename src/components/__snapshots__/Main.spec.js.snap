// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders StyledMain 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 8px;
}

.c0 > div {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 8px;
}

<div
  className="c0"
/>
`;

exports[`Main renders without crashing 1`] = `
<styled.div>
  <OrderList
    orders={
      Array [
        Object {
          "id": 1,
          "price": 123,
          "quantity": 10,
        },
      ]
    }
    title="Buy"
  />
  <OrderList
    orders={
      Array [
        Object {
          "id": 2,
          "price": 233,
          "quantity": 5,
        },
      ]
    }
    title="Sell"
  />
  <withState(withHandlers(MatchList))
    matches={
      Array [
        Object {
          "buy": Object {
            "id": 4,
            "price": 60,
            "quantity": 6,
          },
          "sell": Object {
            "id": 3,
            "price": 50,
            "quantity": 3,
          },
          "time": 123,
        },
      ]
    }
  />
</styled.div>
`;
